Code/Framework/AzCore/Tests/Math/MathUtilsTests.cpp

```diff
@@ -180,6 +180,25 @@ namespace UnitTest
         DivideAndRoundUp(
             static_cast<uint64_t>((AZStd::numeric_limits<uint64_t>::max() / 2) + 1),
             static_cast<uint64_t>((AZStd::numeric_limits<uint64_t>::max() / 2) + 1));
+#ifdef AZ_DEBUG_BUILD
+        // AZ_MATH_ASSERT only asserts during debug builds, so expect an assert here
+        AZ_TEST_STOP_TRACE_SUPPRESSION(1);
+#else
+        AZ_TEST_STOP_TRACE_SUPPRESSION(0);
+#endif
```

Comment:
Instead of having the `#ifdef AZ_DEBUG_BUILD` blocks, `AZ_TEST_STOP_TRACE_SUPPRESSION_NO_COUNT` can be used

Reply:
The idea is that it should be asserting in debug builds, but not during profile builds. Using NO_COUNT would omit the validation that the assert occurs on invalid input.